client/src/client.rs

```diff
@@ -1106,6 +1105,25 @@ impl RpcApi for Client {
         log_response(cmd, &resp);
         Ok(resp?.into_result()?)
     }
+
+    fn version(&self) -> Result<usize> {
+        if let Some(version) = *self.version.read().unwrap() {
+            return Ok(version);
+        }
+
+        let version = get_version(self)?;
+        *self.version.write().unwrap() = Some(version);
+        Ok(version)
+    }
+}
```

Comment:
An alternative is to use `std::sync::Once` + `std::cell::UnsafeCell` (or `once_cell` https://docs.rs/once_cell/1.7.2/once_cell/sync/struct.OnceCell.html)

Reply:
(There's a potential "race" in this RW approach, but the worse case is that `get_version` gets called more than once, it will return the same value anyway)